modules/statistics-service-api/src/main/java/org/opencastproject/statistics/api/StatisticsUtil.java

```diff
@@ -67,6 +68,12 @@ private StatisticsUtil() {
           result.add(currentWeekStart); // No min, max here, because influx uses previous monday as bucket label
           localStart = localStart.plusDays(7).plusDays(daysOff).withHour(0).withMinute(0).withSecond(0).withNano(0);
           break;
+        case HOURLY:
+          final Instant currentHourStart = ZonedDateTime.of(localStart.withMinute(0).withSecond(0).withNano(0), zoneId)
+              .toInstant();
+          result.add(Ordering.natural().min(to, Ordering.natural().max(from, currentHourStart)));
+          localStart = localStart.plusHours(1).withMinute(0).withSecond(0).withNano(0);
+          break;
```

Comment:
Here we set `currentHourStart` to  an Instant with minute `0`. The idea of the min max is to not run out of the bounds defined by `from` and `to` in case `from` and/or `to` have minute != 0. E.g. in case of from=...T08:30:00Z we don't want to get the 08:00 Instant. However, this is more or less theoretical since the frontend should only provide full-hour Instants. (I am not sure what happens in the greatly feared timezone "America/St_Johns", though...)

Reply:
My thought process is something like this: We make sure that localStart is before localEnd in the loop, and currentHourStart can only ever become less by setting minutes, seconds etc to 0, not more, right? So while currentHourStart could become less than from, I can't imagine a case where currentHourStart could ever be bigger than to. Or am I missing something?

Reply:
Ah now I see it. You are right. `min` is not required, but `max` is. I removed the `min`.